src/clients/Deluge.ts

```diff
@@ -233,32 +247,53 @@ export default class Deluge implements TorrentClient {
 	/**
 	 * if Label plugin is loaded, adds (if necessary)
 	 * and sets the label based on torrent hash.
+	 * @param newTorrent the searchee of the newTorrent
+	 * @param label the destination label for the newTorrent/searchee
 	 */
-	private async setLabel(
-		name: string,
-		infoHash: string,
-		label: string,
-	): Promise<void> {
+	private async setLabel(newTorrent: Searchee, label: string): Promise<void> {
+		let setResult: Result<void, ErrorType>;
 		if (!this.isLabelEnabled) return;
 		try {
-			const setResult = await this.call<void>("label.set_torrent", [
-				infoHash,
-				label,
-			]);
-			if (setResult.error?.code === DelugeErrorCode.RPC_FAIL) {
+			const getCurrentLabels = await this.call<string[]>(
+				"label.get_labels",
+				[],
+			);
+			if (getCurrentLabels.isErr()) {
```

Comment:
It shouldn't be necessary, given the change I made to check which labels already exist. Previously we were relying on that response to be the way to determine if it needs to create the label.

It now does this correctly rather than dependence on a failure.

Reply:
sounds good.